app/controllers/hub/source_params_controller.rb

```diff
@@ -0,0 +1,42 @@
+module Hub
+  class SourceParamsController < ApplicationController
+    include AccessControllable
+    before_action :require_sign_in
+    authorize_resource :vita_partner, parent: false, only: :create
+
+    def create
+      vita_partner = VitaPartner.find(params[:vita_partner_id])
+      @form = form_class.new(vita_partner, permitted_params)
+      if @form.valid?
+        @form.save!
+        @success_message = I18n.t("hub.source_params.success", code: @form.code, name: vita_partner.name)
+      else
+        flash.now[:alert] = @form.error_summary
+      end
+      respond_to do |format|
+        format.js
+      end
+    end
+
+    def destroy
+      @source_param = SourceParameter.find_by(id: params[:id])
+
+      unless @source_param.present?
```

Comment:
In the case that there is not a valid id present, that means that you probably weren't using our interface in order to make the request -- you instead were attempting to go to the destroy url directly or some other way of making a delete request to that url. In that case, ejecting with an html response and redirect over js is because we can't do anything legitimate with the data you've provided and it's likely you aren't even looking at our interface.

Reply:
It would be just as valid to raise an error or something i guess